### DISCUSSION TITLE
MeshShader Support?

### DISCUSSION THREAD

**Discussion by albert8271** [vsg-dev/VulkanSceneGraph#856] [Q&A] [answered]
I have run the 'vsgmeshshader' in the example, but the result is follow:

mesh_features.meshShader = 1
mesh_features.taskShader = 1
barycentric_features.fragmentShaderBarycentric = 1
mesh_properites.maxTaskWorkGroupTotalCount = 2147483647
mesh_properites.maxMeshSharedMemorySize = 0
mesh_properites.maxMeshPayloadAndSharedMemorySize = 0
mesh_properites.maxMeshOutputMemorySize = 0
mesh_properites.maxMeshOutputVertices = 0
mesh_properites.maxMeshOutputPrimitives = 0
Using vsg::DrawMeshTasks
Error: vsg::Device::create(...) failed to create logical device. VkResult = -7

My environment is :  (GPU) Nvidia RTX 1650 and VS2019 , and the other examples run normal (except vsgraytracing).
The RTX 1650 support  dont't meshshader?

thank you!

**Comment by vsg-dev** [comment]
I had mesh shaders working on my RTX 1650's quite a while back.  Since then we've moved from the NVidia RTX mesh shader extensions to the Khronos mesh shaders and have tested with my RTX 2060 but haven't put back in the 1650's.  I am just using my AMD 5700G's onboard GPU right now so can't test right away to double check current status.

Form the output it would look like mesh shaders are supported and should work, but the vkDevice isn't being created due to VkResult -7 which is VK_ERROR_EXTENSION_NOT_PRESENT.  Perhaps the meshshaders example isn't enabling the mesh shader correctly, or perhaps the driver is an issue.

I have Kubuntu 22.04 on my system and will try out my 2060 and 1650 cards today before the 1.0.7 release to see anything needs changing to the VSG or vsgExamples.

**Reply by vsg-dev** [reply, marked as answer]
I have just tried my Geforce 2060 and 1650 cards on my Kubuntu 22.04 system with NVidia 525.105.17 drivers and both run vsgmeshshaders example correctly without any warnings.

Older NVidia drivers didn't support Khronos Mesh Shaders so you may need to update drivers to get support for them.

**Reply by albert8271** [reply]
thank you!
I updated the driver to 535.98, the vsgmeshshader run correctly!
